### PR TITLE
fix(core): fix for Windows platform

### PR DESCRIPTION
**Pull request by StephenSorriaux**
Fixes https://github.com/python-zk/kazoo/issues/679

## Why is this needed?
This fixes an issue with the Windows platform that appeared in the latest 2.9.0 release. This also adds a very basic testing for the Windows platform (only 1 job, for the latest version of Python and the latest version of ZK).

## Proposed Changes
  - remove the `os.fstat(fd)` check that is invalid on Windows platform
  - adjust `test_selector_select` test
  - add a single job to test Windows platform (`gevent` seems broken on Windows so... skipped it): only 1 job because I don't feel like we need to re-test every version of Python/ZK for Windows

## Does this PR introduce any breaking change?
Nah.

### CODE REVIEW

File: kazoo/handlers/utils.py [hunk 1]
```diff
@@ -354,7 +354,11 @@ def fileobj_to_fd(fileobj):
             raise TypeError("Invalid file object: " "{!r}".format(fileobj))
     if fd < 0:
         raise TypeError("Invalid file descriptor: {}".format(fd))
-    os.fstat(fd)
+
+    # XXX: needed?
+    # selectors_module.register handle fileobject, do we need the use of fileobj_to_fd()?
+    # is this test valid for Windows?
+    # os.fstat(fd)
```
Comment (StephenSorriaux) [inline kazoo/handlers/utils.py:361]: Hello @JetDrag,

Sorry to summon you here. This `os.fstat(fd)` call is generating crashes on Windows because the handle is, it seems, invalid. With this call commented out, the crash does not appear (and tests are running fine).
Since you did this implementation, I was wondering if:
- there was a reason using the `fileobj_to_fd()` method instead of directly giving the fileobj to `selectors_module.register()` (which seems to be capable of handling fileobj directly)?
- you know if the `os.fstat(fd)` test is valid for Windows?

Thank you.

Comment (ceache) [reply kazoo/handlers/utils.py:361]: I second this.
This `os.fstat()` does not do anything. 
Actually, python3 handles `os.stat()` on FD or file just fine.

Comment (ceache) [reply kazoo/handlers/utils.py:361]: https://docs.python.org/3/library/os.html#os.stat

Comment (JetDrag) [reply kazoo/handlers/utils.py:361]: Hi, @StephenSorriaux @ceache 

The function `fileobj_to_fd` is same as `_fileobj_to_fd` function in `selectors` module  almost except the last second line `os.fstat(fd)`.

I did this implementation to simulate the raw select behavior via test case `test_selectors_select`, which is modified from Python official test cast `test_select`.

If I don't do that, the test cast `test_errno` in the `test_selectors_select` will fail, because the exception (OSError with errno 22, EINVAL) raise by  `selectors._BaseSelectorImpl` module don't meet the test case requirement：raise OSError with errno 9 (EBADF) when select meet a closed fd.

Maybe we cloud make some other simulate implementation to make those spec work on the Windows?

Comment (ceache) [reply kazoo/handlers/utils.py:361]: https://github.com/python/cpython/blob/main/Lib/test/test_select.py#L32  `test_errno()` for reference.

I do not think we care about this behavior in Kazoo.
Actually, the err conditions with selector when selecting "bad file descriptors" are varied depending on platform and selector implementations (there are threads about this on the Python bug report).
This is here trying to make sure that selectors implementations are all equivalente to `select()` behavior and we really can't do that.

IMHO. the only thing we need to do is ensure how Kazoo reacts to a closed FD.
In other words, be it `EINVAL` or `EBADF` or other, we should raise an uniform Kazoo error (with Python3, we could pass up the original, OS/Implementation dependent Exception).

What do you think?

**Comment by ceache** [comment]
@StephenSorriaux  is this a draft or ready for review yet?

**Comment by StephenSorriaux** [comment]
@ceache Not really, it is still a draft. I'm just too used to gitlab, which auto marks a PR as draft if "draft" is in the title.

File: kazoo/handlers/utils.py (hunk 1 above)
Comment (JetDrag) [reply kazoo/handlers/utils.py:361]: It maybe the best solution.

Comment (StephenSorriaux) [reply kazoo/handlers/utils.py]: Thank you both for your thoughts. As suggested, I tried to unify the exceptions that can be raised and catch that in the `test_errno()` testcase. That may be too naive, not sure really, so I would love your feedback on this.

**Comment by jeffwidman** [comment]
What's needed to move this forward? It'd be nice to get this resolved for our Windows users.

**Comment by StephenSorriaux** [comment]
I was waiting for feedbacks from @JetDrag / @ceache but I will go ahead and just finish it so that it can be reviewed.

**Review by ceache — DISMISSED** [review]
Thanks!

**Review by jeffwidman — APPROVED** [review]
Looks fine to me, with the caveat that I haven't actually written/run python on 🪟 in years.

**Comment by JetDrag** [comment]
Excellent work！ @StephenSorriaux

File: .github/workflows/testing.yml [hunk 2]
```diff
@@ -111,3 +110,54 @@ jobs:
 
       - name: Publish Codecov report
         uses: codecov/codecov-action@v3
+
+  test_windows:
+    needs: [validate]
+    name: Windows - Test Python 3.10, ZK 3.7.1
```
[the remark this answers is not in the document]
Comment (StephenSorriaux) [reply .github/workflows/testing.yml]: Thanks, I updated some of the names

File: .github/workflows/testing.yml [hunk 3]
```diff
@@ -111,3 +110,54 @@ jobs:
...
+    steps:
+      - name: Handle the code
+        uses: actions/checkout@v3
+
+      - name: "Set up Python 3.10"
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+
+      - name: Handle pip cache
```
[the remark this answers is not in the document]
Comment (StephenSorriaux) [reply .github/workflows/testing.yml:123]: I think it could, but I am not sure this is something I want to try in this PR

**Comment by StephenSorriaux** [comment]
@ceache @jeffwidman Sorry to resurrect this only now, would appreciate another review if possible and finally merge this.

**Review by jeffwidman — APPROVED** [review]
LGTM, with the caveat that I don't run Windows. But I think the additional confidence of the basic sanity test is enough to be comfortable merging this.

Thank you for your hard work on this!
